app/assets/javascripts/services/miq_service.js

```diff
@@ -140,6 +140,20 @@ ManageIQ.angular.app.service('miqService', ['$timeout', '$document', '$q', 'API'
     return $q.reject(e);
   };
 
+  this.resetClicked = function(form, vm) {
+    vm[vm.modelName] = angular.copy(vm.modelCopy);
+    form.$setPristine(true);
+    miqService.miqFlash("warn", __("All changes have been reset"));
+  };
+
+  this.cancelClicked = function(_form, vm) {
+    if (vm.newRecord) {
+      miqService.redirectBack(sprintf(__('Creation of new %s was canceled by the user.'), vm.entity), 'warning', vm.redirectUrl);
+    } else {
+      miqService.redirectBack(sprintf(__('Edit of %s \"%s\" was canceled by the user.'), vm.entity, vm.entityName), 'warning', vm.redirectUrl);
+    }
+  };
+
```

Comment:
Not really sure if `miqService` or any service for that matter should be responsible for any button based UI interactions.

I'm really of the school of thought that button based operations should go hand-in-hand with a form based component, and that's where it should be ideally handled.